src/porepy/grids/mdg_generation.py

```diff
@@ -401,14 +401,12 @@ def _validate_args(
         domain: Union[pp.Domain, None] = _retrieve_domain_instance(fracture_network)
         if dim == 2:
             assert isinstance(fracture_network, FractureNetwork2d)
-            _, edges_deleted = fracture_network.impose_external_boundary(domain)
-            sz = edges_deleted.size
+            _, fractures_deleted = fracture_network.impose_external_boundary(domain)
         elif dim == 3:
             assert isinstance(fracture_network, FractureNetwork3d)
-            fractures_deleted = fracture_network.impose_external_boundary(domain)
-            # Take note of deleted fractures
-            sz = fractures_deleted.size
-        if sz > 0:
+            _, fractures_deleted = fracture_network.impose_external_boundary(domain)
```

Comment:
Here, we access the new return data. Previously, the saved fractures mistakenly were assigned to a variable `fractures_deleted` here.